### PR TITLE
[hf inference] ASR remote inference model parser impl

### PR DESCRIPTION
**Pull request by Ankush-lastmile**
[hf inference] ASR remote inference model parser impl

Implementation of the HuggingFaceAutomaticSpeechRecognition Model parser using the inference endpoint to run inference. Python API takes in bytes as well as path, skip binary for now.

Very similar to #1018

## Testplan
Screenshot 2024-01-24 at 10 37 05 PM

1. Temporarily add model parser to Gradio Cookbook model parser registry.
```
    asr = HuggingFaceAutomaticSpeechRecognitionRemoteInference()
    AIConfigRuntime.register_model_parser(
        asr, asr.id()
    )
```

2. run AIConfig Edit on Gradio example

`python3 -m 'aiconfig.scripts.aiconfig_cli' edit --aiconfig-path=cookbooks/Gradio/huggingface.aiconfig.json --parsers-module-path=cookbooks/Gradio/hf_model_parsers.py --server-mode=debug_servers`

### CODE REVIEW

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:80]:
```suggestion
                parser = HuggingFaceAutomaticSpeechRecognitionRemoteInference()
```

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:142]:
```suggestion
        audio: Union[Path, str, bytes, BinaryIO] = data["audio"]
```

Should fix the name and typing here

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:148]: here and throughout
```suggestion
        if isinstance(audio, Path):
```

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:212]: It only supports single audio input, based on https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L304, right?

If so, please update the comment and variable handling here to not be an array

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:219]: Ah, ok, that's what I was thinking. Instead of doing this, we should just make the validate_and_retrieve function return a single value, not array

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (rholinshead) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py:310]: Based on your comment on the image_2_text one, maybe this should raise a ValueError instead?

**Review by rholinshead — CHANGES_REQUESTED** [review]
Some minor changes, mostly from copy/paste

**Comment by Ankush-lastmile** [comment]
- Fixed a couple of nits on comments
- Renamed any references of `image` to `audio`
- Refactored `validate_and_retrieve_audio_from_attachments` to also validate only one audio attachment

## Testplan,

Same as in original pr description, output had the same output so omitting the screenshot

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: Yep, updated. Had previously copy pasted what you had for consistency.

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: updated. refactpred validate_and_retrieve  returns a single value, not an array

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: updated

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: updated

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: updated comment

**Review by rholinshead — APPROVED** [review]
Accepting to unblock, but please update the mimetype validation before landing

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/automatic_speech_recognition.py]: ah I see what you mean. Technically this doesn't break anything yet but will be needed.

Updated, thanks for catching

**Comment by Ankush-lastmile** [comment]
update the mimetype validation to check `audio`, not `audio/`

**Pull request merged**
